sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -449,54 +515,137 @@ static RunQueryRequest makeRequest(Query query, @Nullable String namespace) {
 
     @Override
     public void validate(PBegin input) {
-      checkNotNull(getProjectId(), "projectId");
-      checkNotNull(getQuery(), "query");
+      checkNotNull(getProjectId(), "projectId ValueProvider");
+
+      if (getProjectId().isAccessible()) {
+        checkNotNull(getProjectId().get(), "projectId");
```

Comment:
Even today this problem exists as the user is free to pass any `projectId`, but we only check for null? 

I get this message from cloud when I try to create a new project `Project name has invalid characters. Enter letters, numbers, quotes, hyphens, spaces or exclamation points.` But if we want to validate it correctly, then we want to check if the `project` exists by making a call to the cloud?